src/test/test-monitors.c

```diff
@@ -354,5 +354,39 @@ custom_entry_type(void)
     sol_monitors_clear(ms);
 }
 
+DEFINE_TEST(infinite_loop_test);
+
+static void
+infinite_loop_cb(const void *data)
+{
+    static uint8_t counter = 0;
+    struct sol_monitors *ms = (struct sol_monitors *)data;
```

Comment:
no need to cast... and the function should be `(void *data)`, no const in there.